sql/indicator_functions.sql

```diff
@@ -280,7 +280,42 @@ CREATE FUNCTION PECG_Indicator17(
     p_startAge INT(11),
     p_endAge INT (11),
     p_includeEndAge TINYINT(1),
-    p_gender VARCHAR(1)) RETURNS INT(11)
+    p_gender VARCHAR(1),
+    p_monthOffset INT(11)) RETURNS INT(11)
+    DETERMINISTIC
+BEGIN
+    DECLARE result INT(11) DEFAULT 0;
+
+SELECT
+    COUNT(DISTINCT pat.patient_id) INTO result
+FROM
+    patient pat
+WHERE
+    patientGenderIs(pat.patient_id, p_gender) AND
+    patientAgeWhenRegisteredForHivProgramIsBetween(pat.patient_id, p_startAge, p_endAge, p_includeEndAge) AND
+    patientHasEnrolledIntoHivProgramBefore(pat.patient_id, p_endDate) AND
+    patientHasStartedARVTreatmentBefore(pat.patient_id, p_startDate) AND
+    patientDidntCollectARV(pat.patient_id, p_startDate, p_endDate, 0, 0) AND
+    patientHasScheduledAnARTAppointment(pat.patient_id, p_startDate, p_endDate, 0) AND
+    patientIsNotDead(pat.patient_id) AND
+    patientIsNotLostToFollowUp(pat.patient_id) AND
+    patientIsNotTransferredOut(pat.patient_id);
+
+    RETURN (result);
+END$$ 
+DELIMITER ;
+
+DROP FUNCTION IF EXISTS PECG_Indicator18;
+
+DELIMITER $$
+CREATE FUNCTION PECG_Indicator18(
+    p_startDate DATE,
+    p_endDate DATE,
+    p_startAge INT(11),
+    p_endAge INT (11),
+    p_includeEndAge TINYINT(1),
+    p_gender VARCHAR(1),
+    p_monthOffset INT(11)) RETURNS INT(11)
```

Comment:
@IipperMangesh the parameter `p_monthOffset` is not needed for the function `PECG_Indicator18` (as it is hardcoded when calling `patientDidntCollectARV` and `patientHasScheduledAnARTAppointment`)